core/arch/arm/kernel/kern.ld.S

```diff
@@ -323,13 +331,18 @@ SECTIONS
 
 	.rodata_init : {
 #include <rodata_init.ld.S>
-
+#ifndef CFG_CORE_ASLR
 		. = ALIGN(8);
 		KEEP(*(SORT(.scattered_array*)));
-
-		. = ALIGN(8);
-		__rodata_init_end = .;
+#endif
 	}
+#ifdef CFG_CORE_ASLR
+	.data.rel.ro_init : ALIGN(8) {
+		KEEP(*(SORT(.scattered_array*)));
+	}
+#endif
+	. = ALIGN(8);
+	__rodata_init_end = .;
```

Comment:
Good catch ;-) This is actually fixed by later commit "core: pager: export __{text,rodata}_{init,pageable}_{start,end}". At this point it doesn't matter since the attestation PTA doesn't exist yet.
Still I suppose it would make more sense to introduce `__ro_and_relro_data_init_end` at the same time as the new relro section. I will fix that.